pandas/core/frame.py

```diff
@@ -3909,8 +3909,8 @@ def rename(self, *args, **kwargs):
         copy : boolean, default True
             Also copy underlying data
         inplace : boolean, default False
-            Whether to return a new DataFrame. If True then value of copy is
-            ignored.
+            If True, perform the operation in-place, modifying this DataFrame,
+            and return None.
```

Comment:
I don't think this is technically correct. The name is misleading, but using `inplace` doesn't necessarily mean that we're modifying the `DataFrame`, as we can be returning a copy.

I think the previous description is more accurate, except that it sounds like `inplace=True` will return a new `DataFrame`.

Reply:
If you have a `DataFrame` named `df`, using the memory in the address 1000-2000, and you do `df.operation(inplace=True)`, `df` will be modified, but the data can be now in the memory address 3000-4000.

So, technically speaking, the operation is not in place, as the result has been copied to a different location (even if the reference `df` is reassigned to the new memory).

That's why I don't like your description, as for a user who doesn't care about the memory is perfectly fine and correct. But if you think about the memory, you description is not really correct.

Reply:
@nmusolino there are about 2 inplace operations that *actually* operate inplace. all others, simply do an internal copy, do the op, then assign back to the top level data. inplace has never been a real operation in pandas, nor will it ever be. This is why we are deprecating it. You can certainly give a longer explanation, but discouraging usage would be good here.

Reply:
@jreback are we ready to deprecate all `inplace` parameters? A quick grep shows around 60 methods with an `inplace` parameter, so it's not so much work to start raising warnings (and update the descriptions).

Or do we prefer to wait?

Reply:
@datapythonista :

See #16529 - I think we were planning to wait until 1.0, though can always push up.

Reply:
Thanks for pointing out the issue @gfyoung, I'll continue the conversation there.

Reply:
@datapythonista, @jreback , I understand that these operations may involve copies.  I  wrote earlier that "internal data structures (e.g. block manager or index) might be copied even when inplace is True."  

I think the distinction you are trying to draw is that a `rename` or `fillna` operation does not necessarily overwrite individual elements in place; is that your point?  As a simple example, I seem to recall that Index objects are generally immutable, so any operation that changes the index would necessarily replace the index.

To be clear, however, the methods are modifying the dataframe.

How about this description?
```
    inplace : boolean, default False
        If true, rename index labels in this DataFrame and return None, 
        without creating a new DataFrame.
```

Or a simpler, more generic version:
```
    inplace : boolean, default False
        If true, modify this DataFrame and return None; does not create a 
        new DataFrame.
```